examples/quickrefl/core/model/realdatamodel.cpp

```diff
@@ -96,6 +97,30 @@ void RealDataModel::addDataToGroup(DataGroupItem* data_group, RealDataStruct& da
     graph->setDataItem(data);
 }
 
+//! Insert the data into the group item
+void RealDataModel::removeAllDataFromNode(DataCollectionItem* data_node)
+{
+    for (auto item : data_node->children()) {
+        removeItem(item->parent(), item->parent()->tagRowOfItem(item));
+    }
+}
+
+//! Insert the data into the group item
+void RealDataModel::removeDataFromNode(std::vector<ModelView::SessionItem*> item_to_remove)
+{
+    for (auto item : item_to_remove) {
+        if (auto group_item = dynamic_cast<DataGroupItem*>(item)) {
+            for (auto temp_item : group_item->children()) {
+                if (auto sub_item = dynamic_cast<GraphItem*>(temp_item))
+                    removeDataFromGroup(sub_item);
+            }
+            removeItem(group_item->parent(), group_item->parent()->tagRowOfItem(group_item));
+        } else if (auto subitem = dynamic_cast<GraphItem*>(item)) {
+            removeDataFromGroup(subitem);
+        }
+    }
+}
```

Comment:
At some point would be nice to cover RealDataModel with tests. It starts to get complicated logic on board.